### DISCUSSION TITLE
How to get Content Items from Keyword in Orchard Search

### DISCUSSION THREAD

**Discussion by Syed-Hasnain-Askari** [OrchardCMS/OrchardCore#12278] [Q&A] [answered]
Hey,

I'm working on Orchard Lucene Search, I want to search content Items based on my keyword. for example, there's is 8 Content Item of "Prime" so when I type Prime so all the Prime will be show.

This is my query which is work fine.
`{
  "query": {
    "bool": {
      "must" : {
          "term" : { "Content.ContentItem.ContentType" : "Software" }
      },
      "filter": [
        { "term": { "Content.ContentItem.Published" : "true" }},
        { "wildcard": { "Content.ContentItem.DisplayText" : "Prime*" }}
      ]
    }
  }
}`
![image](https://user-images.githubusercontent.com/47834896/186918597-e3269498-9625-4488-9cc8-40a267cae4de.png)

but I don't know use this query into the Lucene controller. I will be grateful for any help you can provide.

**Comment by saleemhaq** [comment]
@Skrypt please help or advise on above

**Reply by Skrypt** [reply]
https://docs.orchardcore.net/en/dev/docs/reference/modules/Queries/#query_1

**Reply by saleemhaq** [reply]
> https://docs.orchardcore.net/en/dev/docs/reference/modules/Queries/#query_1
@Skrypt can you please advise me on below discussion:
https://github.com/OrchardCMS/OrchardCore/discussions/12280

**Reply by saleemhaq** [reply]
> https://docs.orchardcore.net/en/dev/docs/reference/modules/Queries/#query_1

@Skrypt how we add multiple content type like interface, Apps and others like we have currently software

![image](https://user-images.githubusercontent.com/5202291/187527731-377c79c6-ac58-4aa2-96e9-691f564cf706.png)

Can you share with us the example @ns8482e @Skrypt

**Reply by ns8482e** [reply]
As your query you shared above works, give it a name, replace the  search text with parameter and save the query.

using orchard helper method`ContentQueryAsync(string queryName, IDictionary<string, object> parameters) ` execute query by passing queryname and parameters from your search box

**Reply by saleemhaq** [reply]
> replace the search text with parameter

@ns8482e , can you share with me the example how i pass the  search text with parameter @Skrypt

**Reply by ns8482e** [reply]
@saleemhaq you can use sample query shared by @Skrypt  and replace your search keyword "Prime" with variable. 
```
 "wildcard": {
          "Content.ContentItem.DisplayText": "{{ searchVar }}*"
        }
```

To run query with parameter you can pass parameter json ` { searchVar  : 'Prime' }`

**Reply by saleemhaq** [reply]
> @saleemhaq you can use sample query shared by @Skrypt and replace your search keyword "Prime" with variable.
> 
> ```
>  "wildcard": {
>           "Content.ContentItem.DisplayText": "{{ searchVar }}*"
>         }
> ```
> 
> To run query with parameter you can pass parameter json ` { searchVar : 'Prime' }`

Please help in below:

![RequiredSearch](https://user-images.githubusercontent.com/5202291/191093923-55876430-1835-4eeb-ab72-114d49fe6540.png)

@ns8482e @Skrypt

**Reply by Skrypt** [reply]
Use Content.ContentItem.DisplayText.Normalized then for that matter.

**Reply by ns8482e** [reply]
it happens a lot to me,  Sometimes the  solution is in front of you but can’t recognize  😀

**Comment by saleemhaq** [comment, marked as answer]
I want to add more content type like software and others
like csv values "Software","Interfaces","Other"
When modify above query and add values like above in csv format so it give error see below query:

{ "query": { "bool": { "must" : { "term" : { "Content.ContentItem.ContentType" : "Software","Interfaces","Other" } }, "filter": [ { "term": { "Content.ContentItem.Published" : "true" }}, { "wildcard": { "Content.ContentItem.DisplayText" : "Prime*" }} ] } } }

where interface and other is differ content type, when execute this query it give error. @ns8482e please advise.

**Reply by Skrypt** [reply]
You need to make use of "should" which is a logical "OR" instead of "must" which is a logical "AND".
```
{
  "query": {
    "bool": {
      "should": [
        {
          "term": {
            "Content.ContentItem.ContentType": "Software"
          }
        },
        {
          "term": {
            "Content.ContentItem.ContentType": "Interfaces"
          }
        },
        {
          "term": {
            "Content.ContentItem.ContentType": "Other"
          }
        }
      ]
    },
    "filter": [
      {
        "term": {
          "Content.ContentItem.Published": "true"
        }
      },
      {
        "wildcard": {
          "Content.ContentItem.DisplayText": "Prime*"
        }
      }
    ]
  }
}
```

**Reply by saleemhaq** [reply]
@ns8482e @Skrypt please help me in below:

![RequiredSearch](https://user-images.githubusercontent.com/5202291/191217955-f816ae65-f750-40f0-97e2-256981b93813.png)

Please advise me on this:

**Reply by Skrypt** [reply]
Use Content.ContentItem.DisplayText.Normalized then for that matter.

**Reply by Syed-Hasnain-Askari** [reply]
Thankyou for your kind feedback! @Skrypt @ns8482e
